src/server/db_slice.cc

```diff
@@ -998,16 +998,16 @@ bool DbSlice::Acquire(IntentLock::Mode mode, const KeyLockArgs& lock_args) {
   bool lock_acquired = true;
 
   if (lock_args.args.size() == 1) {
-    string_view key = KeyLockArgs::GetLockKey(lock_args.args.front());
-    lock_acquired = lt.Acquire(key, mode);
-    uniq_keys_ = {key};  // needed only for tests.
+    LockTag tag = GetLockTag(lock_args.args.front());
+    lock_acquired = lt.Acquire(tag, mode);
+    uniq_keys_ = {tag.str};  // needed only for tests.
   } else {
     uniq_keys_.clear();
 
     for (size_t i = 0; i < lock_args.args.size(); i += lock_args.key_step) {
-      string_view s = KeyLockArgs::GetLockKey(lock_args.args[i]);
-      if (uniq_keys_.insert(s).second) {
-        lock_acquired &= lt.Acquire(s, mode);
+      LockTag tag = GetLockTag(lock_args.args[i]);
+      if (uniq_keys_.insert(tag.str).second) {
+        lock_acquired &= lt.Acquire(tag, mode);
```

Comment:
because with the implicit operator, the existing expressions like "string_view key = tag" compiled.
I did not want them to compile

Reply:
Why not? As long as table accepts only LockTag, it's not a problem. You can get a string_view but you can't do anything useful with it

Reply:
made more changes. I prefer explicit conversion because it allowed me to track all the problematic conversions of the current code back to string_view that I wanted to avoid. I prefer to cover all the cases where we operate on tags with LockTag